### PR TITLE
Fix: minimize floating point errors in AudioBufferSourceNode current_time calculation

### PR DESCRIPTION
**Pull request by b-ma**
Fixes the crashing the `the-audio-api/the-audiobuffersourcenode-interface/sample-accurate-scheduling.html` wpt test

### CODE REVIEW

File: src/node/audio_buffer_source.rs [hunk 1]
```diff
@@ -649,19 +649,19 @@ impl AudioProcessor for AudioBufferSourceRenderer {
         let mut playback_infos = [None; RENDER_QUANTUM_SIZE];
 
         // compute position for each sample and store into `self.positions`
-        for playback_info in playback_infos.iter_mut() {
+        for (i, playback_info) in playback_infos.iter_mut().enumerate() {
+            let current_time = block_time + i as f64 * dt;
```
Comment (b-ma) [inline src/node/audio_buffer_source.rs:653]: This is actually the main change

File: src/node/audio_buffer_source.rs [hunk 2]
```diff
@@ -649,19 +649,19 @@ impl AudioProcessor for AudioBufferSourceRenderer {
...
+
             if current_time < self.start_time
                 || current_time >= self.stop_time
                 || self.render_state.buffer_time_elapsed >= self.duration
             {
-                *playback_info = None;
-                current_time += dt;
```
Comment (b-ma) [inline src/node/audio_buffer_source.rs:658]: Rather than that

**Review by orottier — APPROVED** [review]
Thanks nice fix - could you have at my nitpick?

**Comment by b-ma** [comment]
Done!

**Pull request merged**